test/abstract_syntax_tree.coffee

```diff
@@ -10,207 +10,595 @@ getExpressions = (code) ->
-  eq expression.type, 'YieldReturn'
-  eq expression.expression.type, 'Code'
+  testExpression 'yield return ->',
+    type: 'YieldReturn'
+    expression: type: 'Code'
 
 test "AST as expected for AwaitReturn node", ->
-  expression = getExpression 'await return ->'
-  eq expression.type, 'AwaitReturn'
-  eq expression.expression.type, 'Code'
+  testExpression 'await return ->',
+    type: 'AwaitReturn'
+    expression: type: 'Code'
 
 test "AST as expected for Value node", ->
-  expression = getExpression 'for i in [] then i'
-  value = expression.body
-  eq value.type, 'Value'
-  # eq value.value, 'i'
+  testExpression 'for i in [] then i',
+    body:
+      type: 'Value'
+      isDefaultValue: no
+      base: value: 'i'
+      properties: []
+
+  testExpression 'if 1 then 1 else 2',
+    body:
+      type: 'Value'
+      isDefaultValue: no
+    elseBody:
+      type: 'Value'
+      isDefaultValue: no
+
+  # TODO: Figgure out the purpose of `isDefaultValue`. It's not set in `Switch` either.
 
 # Comments aren’t nodes, so they shouldn’t appear in the AST.
 
 test "AST as expected for Call node", ->
-  expression = getExpression 'fn()'
-  eq expression.type, 'Call'
-  variable = expression.variable
-  eq variable.value, 'fn'
-
-  expression = getExpression 'new Date()'
-  eq expression.type, 'Call'
-  eq expression.isNew, yes
-
-  expression = getExpression 'maybe?()'
-  eq expression.type, 'Call'
-  eq expression.soak, yes
-
-  expression = getExpression 'goDo this, that'
-  eq expression.type, 'Call'
-  eq expression.args[0].value, 'this'
-  eq expression.args[1].value, 'that'
+  testExpression 'fn()',
+    type: 'Call'
+    variable: value: 'fn'
+
+  testExpression 'new Date()',
+    type: 'Call'
+    variable: value: 'Date'
+    isNew: yes
+
+  testExpression 'maybe?()',
+    type: 'Call'
+    soak: yes
+
+  testExpression 'goDo this, that',
+    type: 'Call'
+    args: [
+      {value: 'this'}
+      {value: 'that'}
+    ]
+
+  testExpression 'do ->',
+    type: 'Call'
+    do: yes
+    variable: type: 'Code'
+
+  testExpression 'do fn',
+    type: 'Call'
+    do: yes
+    variable:
+      type: 'IdentifierLiteral'
+      value: 'fn'
 
 test "AST as expected for SuperCall node", ->
-  expression = getExpression 'class child extends parent then constructor: -> super()'
-  # TODO
+  testExpression 'class child extends parent then constructor: -> super()',
+    body: base: properties: [
+      value: body: base:
+        type: 'SuperCall'
+    ]
 
 test "AST as expected for Super node", ->
-  # TODO
+  testExpression 'class child extends parent then func: -> super.prop',
+    body: base: properties: [
+      value: body: base:
+        type: 'Super'
+        accessor: type: 'Access'
+    ]
 
 test "AST as expected for RegexWithInterpolations node", ->
-  # TODO
+  testExpression '///^#{flavor}script$///',
+    type: 'RegexWithInterpolations'
+
+  # TODO: Shouldn't there be more info we can check for?
```

Comment:
Could you check for the body of the regex?

Reply:
That's what i wanted to do but it seems the information is missing. Here's the full AST from `getExpressions`:
```js
[
  {
    type: 'RegexWithInterpolations',
    loc: {
      start: {line: 0, column: 0},
      end: {line: 0, column: 22}
    }
  }
]
```

Reply:
Yeah, I think this is the kind of thing that @helixbass is adding in his work.